src/rgw/rgw_sal_motr.cc

```diff
@@ -3066,7 +3203,13 @@ int MotrMultipartWriter::complete(size_t accounted_size, const std::string& etag
   char buf[32];
   snprintf(buf, sizeof(buf), "%08d", (int)part_num);
   p.append(buf);
-  string obj_part_iname = "motr.rgw.object." + head_obj->get_bucket()->get_name() + "." +
+  string tenant_bucket = head_obj->get_bucket()->get_tenant();
+  if (tenant_bucket != "") {
+    tenant_bucket += "$"+head_obj->get_bucket()->get_name();
+  } else {
+    tenant_bucket = head_obj->get_bucket()->get_name();
+  }
```

Comment:
Please, create a static function for this code:

```c
static string& get_bucket_name(string &bucket, string &tenant)
{
  if (tenant != "")
    return tenant + "$" + bucket;
  else
    return bucket;
}
```
and use it everywhere.

Reply:
Done